stl/inc/tuple

```diff
@@ -343,6 +369,32 @@ public:
         tuple(const pair<_First, _Second>&& _Right) noexcept(
             _Tuple_nothrow_constructible_v<tuple, const _First, const _Second>) // strengthened
         : tuple(_Unpack_tuple_t{}, _STD move(_Right)) {}
+
+#ifdef __cpp_lib_concepts
+    template <class _Other, class _Indices = index_sequence_for<_Rest...>>
+    static constexpr bool _Can_construct_from_tuple_like_v = false;
+
+    template <_Tuple_like _Other, size_t... _Indices>
+    static constexpr bool _Can_construct_from_tuple_like_v<_Other, index_sequence<_Indices...>> =
+        _Different_from<_Other, tuple> && !_Is_subrange_v<remove_cvref_t<_Other>>

```

Comment:
Patterns "_`tuple-like`_/_`pair-like`_ but not a `subrange`" appear several times in [P2165R4](https://wg21.link/p2165r4) (also in LWG-3821). Should we have something like `_Tuple_like_without_subrange`/`_Pair_like_without_subrange`?

Reply:
I'm going to leave it as it is for now - there are only 5 occurrences of `!_Is_subrange_v<remove_cvref_t<_Other>>` in `<tuple>` (one of them is part of workaround) and 2 in `<utility>`.

Reply:
I'm now doing this in #4983.